docs/tutorials/quantum-phase-estimation-qctrl.ipynb

```diff
@@ -0,0 +1,737 @@
+   "id": "5019f20e",
+   "metadata": {},
+   "source": [
+    "## Requirements\n",
+    "\n",
+    "Before starting this tutorial, be sure you have the following installed:\n",
+    "- Qiskit SDK v1.4 or later, with visualization support ( `pip install 'qiskit[visualization]'` )\n",
+    "- Qiskit Runtime v0.40 or later (`pip install qiskit-ibm-runtime`)\n",
+    "- Qiskit Functions Catalog v0.9.0 (`pip install qiskit-ibm-catalog`)\n",
+    "- Fire Opal SDK v9.0.2 or later (`pip install fire-opal`)\n",
+    "- Q-CTRL Visualizer v8.0.2 or later (`pip install qctrl-visualizer`)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "360f9871",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39896f05",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit import (\n",
+    "    ClassicalRegister,\n",
+    "    QuantumRegister,\n",
+    ")\n",
+    "from qiskit import QuantumCircuit\n",
+    "import uuid\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import qiskit\n",
+    "from qiskit import qasm2\n",
+    "from qiskit_aer import AerSimulator\n",
+    "from qiskit_ibm_runtime import QiskitRuntimeService\n",
+    "from qiskit_ibm_runtime import SamplerV2 as Sampler\n",
+    "from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager\n",
+    "import qctrlvisualizer as qv\n",
+    "from qiskit_ibm_catalog import QiskitFunctionsCatalog\n",
+    "\n",
+    "plt.style.use(qv.get_qctrl_style())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9abf8775",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set credentials.\n",
+    "token = \"YOUR_IBM_CLOUD_API_KEY\"\n",
+    "instance = \"YOUR_IBM_CRN\"\n",
+    "\n",
+    "catalog = QiskitFunctionsCatalog(\n",
+    "    channel=\"ibm_quantum_platform\", instance=instance, token=token\n",
+    ")\n",
+    "\n",
+    "# Access Function\n",
+    "perf_mgmt = catalog.load(\"q-ctrl/performance-management\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6337a6c2",
+   "metadata": {},
+   "source": [
+    "## Step 1: Map classical inputs to a quantum problem\n",
+    "\n",
+    "We begin by defining the building blocks QPE needed to implement Quantum Phase Estimation. These include the Quantum Fourier Transform (QFT) and its inverse, which translate phase information into measurable bitstrings, along with utility functions to convert between decimal and binary fractions of the target eigenphase. Additional helpers are provided to normalize raw counts into probabilities and reformat distributions, enabling direct comparison between simulation and hardware results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "abcdd911",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def quantum_fourier_transform(qubit_count: int) -> QuantumCircuit:\n",
```

Comment:
For the QPE source code, could we use the previous version (i.e. from [here](https://docs.q-ctrl.com/fire-opal/execute/run-algorithms/improve-the-results-of-quantum-phase-estimation))?